pkg/bgpv1/manager/reconcilerv2/peer_advertisements.go

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package reconcilerv2
+
+import (
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/util/sets"
+
+	"github.com/cilium/cilium/pkg/bgpv1/manager/store"
+	"github.com/cilium/cilium/pkg/bgpv1/types"
+	"github.com/cilium/cilium/pkg/hive/cell"
+	"github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/labels"
+	slim_metav1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
+)
+
+type (
+	// PeerAdvertisements is a map of peer name to its family advertisements
+	// This is the top level map that is returned to the consumer with requested advertisements.
+	PeerAdvertisements       map[string]PeerFamilyAdvertisements
+	PeerFamilyAdvertisements map[v2alpha1.CiliumBGPFamily][]v2alpha1.Advertisement // key is the address family type
+)
+
+type PeerAdvertisementIn struct {
+	cell.In
+
+	Logger          logrus.FieldLogger
+	PeerConfigStore store.BGPCPResourceStore[*v2alpha1.CiliumBGPPeerConfig]
+	AdvertStore     store.BGPCPResourceStore[*v2alpha1.CiliumBGPAdvertisement]
+}
+
+type CiliumPeerAdvertisement struct {
+	logger     logrus.FieldLogger
+	peerConfig store.BGPCPResourceStore[*v2alpha1.CiliumBGPPeerConfig]
+	adverts    store.BGPCPResourceStore[*v2alpha1.CiliumBGPAdvertisement]
+}
+
+func NewCiliumPeerAdvertisement(p PeerAdvertisementIn) *CiliumPeerAdvertisement {
+	return &CiliumPeerAdvertisement{
+		logger:     p.Logger,
+		peerConfig: p.PeerConfigStore,
+		adverts:    p.AdvertStore,
+	}
+}
+
+// GetAdvertisements can be called to get advertisements for given BGPAdvertisementType.
+// Advertisements are selected based on below criteria:
+// Each peer is selected from the BGP node instance configuration. For each peer, the peer configuration is fetched
+// from local store.
+// Peer configuration contains the list of families and the advertisement selector.
+// We iterate over all advertisements ( available from local store ), select only those that match the advertisement
+// selector of the family.
+// Information of peer -> family -> advertisements is returned to the consumer.
+func (p *CiliumPeerAdvertisement) GetAdvertisements(conf *v2alpha1.CiliumBGPNodeInstance, selectAdvertTypes ...v2alpha1.BGPAdvertisementType) (PeerAdvertisements, error) {
```

Comment:
nit: maybe I would name this `GetConfiguredAdvertisements` or similar, to make it obvious that this is returning configuration, not state data.